### PR TITLE
Add reporting guidelines for CoC

### PR DESCRIPTION
**Pull request by Thea-Aldrich**
Adding detailed instructions for reporting suspected violations of the Code of Conduct as per PR #10364 (https://github.com/zephyrproject-rtos/zephyr/pull/10364)

### CODE REVIEW

File: doc/code_of_conduct_reporting_guidelines.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
+.. _Zephyr_Code_of_Conduct_Reporting_Guidelines:
+
+Code of Conduct Reporting Guidelines
+####################################
+
+We are committed to providing members of our community with a safe and productive environment. If you experience or witness violations of the Zephyr Project Code of Conduct please use these reporting guidelines outlined by the Zephyr Projects’ Code of Conduct Working Group. It's used to provide us with as much relevant information as possible when we respond to an issue to make sure we're consistent and fair. If you have questions please contact us at **report@zephyrproject.org**
```
Comment (dbkinder) [inline doc/code_of_conduct_reporting_guidelines.rst]: keep line length < 80 characters when possible

File: doc/code_of_conduct_reporting_guidelines.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Once we've determined our final action, we'll contact the original reporter to let them know what action (if any) we'll be taking. We'll take into account feedback from the reporter on the appropriateness of our response, but we don't guarantee we'll act on it.
+
+Finally, the CoC WG will make a report on the situation to the Zephyr Project Technical Steering Committee (TSC) and Governing Board. The Governing Board may choose to make a public report of the incident.
+
+Reconsideration
+***************
+
+Any of the parties directly involved or affected can request reconsideration of the committee’s decision. To make such a request, contact the Governing Board at **report@zephyrproject.org** with your request and motivation and the Governing Board will review the case.
+
+This document was adapted from the Django Project Code of Conduct Reporting Guidelines. Source: https://www.djangoproject.com/conduct/reporting/__
```
Comment (dbkinder) [inline doc/code_of_conduct_reporting_guidelines.rst]:
```suggestion
This document was adapted from the `Django Project Code of Conduct Reporting
Guidelines <https://www.djangoproject.com/conduct/reporting/>`__.
```

File: doc/code_of_conduct_reporting_guidelines.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+**If you believe anyone is in physical danger, please notify appropriate law enforcement first.** If you are unsure what law enforcement agency is appropriate, please include this in your report.
+
+If you are unsure whether the incident is a violation, or whether the space where it happened is covered by this Code of Conduct, we encourage you to still report it. We would much rather have a few extra reports where we decide to take no action, rather than miss a report of an actual violation. We do not look negatively on you if we find the incident is not a violation. And knowing about incidents that are not violations, or happen outside our spaces, can also help us to improve the Code of Conduct or the processes surrounding it.
+
+Reporting
+*********
+
+In your report please include:
+ * Your contact info (so we can get in touch with you if we need to follow up).
```
Comment (dbkinder) [inline doc/code_of_conduct_reporting_guidelines.rst]: add a blank line before a list, and don't indent the list items:

    In your report please include:

    * Your contact info (so we can get in touch with you if we need
      to follow up).
    * Names (real, nicknames, or pseudonyms) of any individuals
      involved. If there were other witnesses besides you, please
      try to include them as well.

**Review by dbkinder — CHANGES_REQUESTED** [review]
I'll wait for the TSC review of the proposed approach before doing an editor review.  Suggested edits are to fix review issues (really long lines) and formatting (lists).

**Comment by nashif** [comment]
Can this be closed?

**Comment by nashif** [comment]
closing in favour of #10364.

**Pull request closed without merging**